src/snowflake/snowpark/session.py

```diff
@@ -4072,6 +4072,10 @@ def stored_procedure_profiler(self) -> StoredProcedureProfiler:
     def _infer_is_return_table(
         self, sproc_name: str, *args: Any, log_on_exception: bool = False
     ) -> bool:
+        if sproc_name.strip().upper().startswith("SYSTEM$"):
+            # Built-in stored procedures do not have schema and cannot be described
+            # Currently all SYSTEM$ stored procedures are scalar so return false.
+            return False
```

Comment:
is it always illegal to call desc on a system proc?
my major concern is whether this is a breaking change

Reply:
Calling `describe procedure SYSTEM$...` will fail but the function `_infer_is_return_table` has a try-finally protection, so the return value is always for in this case False. However, the failure does show up in query history and is a bad user experience (see JIRA).

Here what we are doing is skipping the describe for system$ procedures so there is no change in behavior and query history stays clean.